src/api/route-services/samples.js

```diff
@@ -192,9 +205,31 @@ class SamplesService {
       await dynamodb.updateItem(params).promise();
     };
 
+    const addSampleIdToExperiments = async () => {
+      const dynamodb = createDynamoDbInstance();
+
+      const marshalledKey = convertToDynamoDbRecord({
+        experimentId,
+      });
+
+      const marshalledData = convertToDynamoDbRecord({
+        ':newSampleId': [newSample.uuid],
+      });
+
+      const params = {
+        TableName: this.experimentsTableName,
+        Key: marshalledKey,
+        UpdateExpression: 'SET sampleIds = list_append(sampleIds, :newSampleId)',
+        ExpressionAttributeValues: marshalledData,
+      };
+
+      await dynamodb.updateItem(params).promise();
```

Comment:
better to surround it into try-catch block

Reply:
I think it's fine to do this. If there is an error, it'll cause the `Promise.all` block on line https://github.com/hms-dbmi-cellenics/api/blob/fda9a046e70e1e75f39f1c548065ba76ce805d62/src/api/route-services/samples.js#L229 to throw an error, which will be received by Express router and returned as an error to the UI. If we want to catch the error and do some clean up (e.g. rollback by deleting the newly created sample id from `projects`, `experiments`, and `samples` table), I think it's best to have a try-catch on the `Promise.all` block. What do you think?

Reply:
Looking at the UI PR, the UI counts on having this data in dynamo (the sample ids) even if the upload didn't fully work out. In this case, I wouldn't be doing a cleanup